packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -327,31 +329,60 @@ Step Functions supports [ECS/Fargate](https://docs.aws.amazon.com/step-functions
 
 [RunTask](https://docs.aws.amazon.com/step-functions/latest/dg/connect-ecs.html) starts a new task using the specified task definition.
 
+#### EC2
+
+The following example runs a job from a task definition on EC2
+
 ```ts
-import * as ecs from '@aws-cdk/aws-ecs';
+const taskDefinition = new ecs.TaskDefinition(stack, 'TD', {
+  compatibility: ecs.Compatibility.EC2,
+});
 
-// See examples in ECS library for initialization of 'cluster' and 'taskDefinition'
+taskDefinition.addContainer('TheContainer', {
+  image: ecs.ContainerImage.fromRegistry('foo/bar'),
+  memoryLimitMiB: 256,
+});
 
-new ecs.RunEcsFargateTask({
+const runTask = new tasks.EcsEc2RunTask(stack, 'Run', {
+  integrationPattern: sfn.IntegrationPattern.RUN_JOB,
   cluster,
   taskDefinition,
-  containerOverrides: [
-    {
-      containerName: 'TheContainer',
-      environment: [
-        {
-          name: 'CONTAINER_INPUT',
-          value: Data.stringAt('$.valueFromStateData'),
-        }
-      ]
-    }
-  ]
+  containerOverrides: [{
+    containerName: 'TheContainer',
```

Comment:
Why not [`ContainerDefinition`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs.ContainerDefinition.html) that's returned from `addContainer()`?

Reply:
the primary reason the overrides are powerful is because they allow selection and definition from state input (i.e. sfn.Data.stringAt(...))

Some of the Step Functions sample documentation leads into it and I've seen from a few users who have shared code that it's being used. If overrides are not specified, the values established the task definition will be used by default.